ui/pages/confirmations/components/confirm/nav/nav.test.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+
+import mockState from '../../../../../../test/data/mock-state.json';
+import { fireEvent, renderWithProvider } from '../../../../../../test/jest';
+import * as Actions from '../../../../../store/actions';
+import configureStore from '../../../../../store/store';
+
+import Nav from './nav';
+
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useDispatch: () => jest.fn(),
+}));
+
+const mockHistoryReplace = jest.fn();
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useHistory: () => ({
+    replace: mockHistoryReplace,
+  }),
+}));
+
+const render = () => {
+  const store = configureStore({
+    metamask: {
+      ...mockState.metamask,
+      pendingApprovals: {
+        testApprovalId: {
+          id: 'testApprovalId',
+          time: 1528133319641,
+          origin: 'metamask',
+          type: 'personal_sign',
+          requestData: {
+            txId: 'testTransactionId',
+          },
+          requestState: {
+            test: 'value',
+          },
+        },
+        testApprovalId2: {
+          id: 'testApprovalId2',
+          time: 1528133319641,
+          origin: 'metamask',
+          type: 'personal_sign',
+          requestData: {
+            txId: 'testTransactionId',
+          },
+          requestState: {
+            test: 'value',
+          },
+        },
+        testApprovalId3: {
+          id: 'testApprovalId3',
+          time: 1528133319649,
+          origin: 'metamask',
+          type: 'personal_sign',
+          requestData: {
+            txId: 'testTransactionId',
+          },
+          requestState: {
+            test: 'value',
+          },
+        },
+      },
+    },
+    confirm: {
+      currentConfirmation: {
+        id: 'testApprovalId2',
+        msgParams: {
+          from: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
+        },
+      },
+    },
+  });
+
+  return renderWithProvider(<Nav />, store);
+};
+
+describe('ConfirmNav', () => {
+  it('should match snapshot', () => {
+    const { container } = render();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders the "Reject all" Button', () => {
+    const { getAllByRole, getByText } = render();
+    const buttons = getAllByRole('button');
+    expect(buttons).toHaveLength(3);
+    expect(getByText('Reject all')).toBeInTheDocument();
+  });
+
+  it('renders button to navigate to previous or next confirmation', () => {
```

Comment:
I see that the buttons are disabled when you are at the beginning or end of the list. Can we test this as well.